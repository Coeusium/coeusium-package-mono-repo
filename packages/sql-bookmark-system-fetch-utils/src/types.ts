export type { Bookmark } from 'sql-bookmark-system';
