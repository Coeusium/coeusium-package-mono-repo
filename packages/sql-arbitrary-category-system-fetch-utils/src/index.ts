export * from './sql-arbitrary-category-system-fetch-utils';
export type { Category, CategorySet } from './types';
