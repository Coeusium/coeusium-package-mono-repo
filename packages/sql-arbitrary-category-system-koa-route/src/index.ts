export * from './sql-arbitrary-category-system-koa-route';
export type { ConnectionInfo, Category, CategorySet } from './types';
