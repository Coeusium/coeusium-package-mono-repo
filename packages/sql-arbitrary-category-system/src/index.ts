export * from './sql-arbitrary-category-system';
export type { ConnectionInfo, Category, CategorySet } from './types';
