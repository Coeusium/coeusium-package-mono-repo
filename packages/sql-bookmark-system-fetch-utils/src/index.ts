export * from './sql-bookmark-system-fetch-utils';
export type { Bookmark } from './types';
