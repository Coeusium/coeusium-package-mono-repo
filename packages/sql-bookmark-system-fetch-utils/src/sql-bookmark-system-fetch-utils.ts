import { Bookmark } from './types';
import { almFetch } from '@almaclaine/fetch-utils';

// Category Set

export async function addBookmark(url) {
  const res = await almFetch(`/api/bookmark?bookmark=${url}`, {
    method: 'post',
  });
  const { id } = await res.json();
  return id as string;
}

export async function getBookmark(id) {
  const res = await almFetch(`/api/bookmark?id=${id}`);
  return (await res.json()) as Bookmark;
}

export async function listBookmarks(page = 0, limit = 20) {
  const res = await almFetch(`/api/bookmark?page=${page}&limit=${limit}`);
  return (await res.json()) as Bookmark[];
}

export async function deleteBookmark(id) {
  const res = await almFetch(`/api/bookmark?id=${id}`, { method: 'delete' });
  await res.json();
}
