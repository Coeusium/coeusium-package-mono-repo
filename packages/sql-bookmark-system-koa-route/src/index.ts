export * from './sql-bookmark-system-koa-route';
export type { ConnectionInfo, Bookmark } from './types';
