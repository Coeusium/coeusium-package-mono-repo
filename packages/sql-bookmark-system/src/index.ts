export * from './sql-bookmark-system';
export type { ConnectionInfo, Bookmark } from './types';
