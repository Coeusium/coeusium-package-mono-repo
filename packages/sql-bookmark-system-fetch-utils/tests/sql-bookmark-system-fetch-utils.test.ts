import * as sqlBookmarkFetchUtils from '../src';

describe('sql-bookmark-fetch-utils', () => {
  it('Has Correct API', () => {
    const keys = Object.keys(sqlBookmarkFetchUtils);
    expect(keys.length).toBe(4);
    expect(keys).toMatchSnapshot();
  });
});
