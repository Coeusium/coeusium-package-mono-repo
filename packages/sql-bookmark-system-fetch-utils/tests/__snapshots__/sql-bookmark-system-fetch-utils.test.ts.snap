// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sql-bookmark-fetch-utils Has Correct API 1`] = `
Array [
  "addBookmark",
  "getBookmark",
  "listBookmarks",
  "deleteBookmark",
]
`;
